src/main/java/ee/carlrobert/codegpt/toolwindow/chat/ChatToolWindowTabPanelEditor.java

```diff
@@ -136,7 +175,47 @@ public void actionPerformed(@NotNull AnActionEvent event) {
+        SelectionModel selectionModel = editor.getSelectionModel();
+        Document document = editor.getDocument();
+        int startLine = document.getLineNumber(selectionModel.getSelectionStart());
+        int endLine = document.getLineNumber(selectionModel.getSelectionEnd());
+        VirtualFile virtualFile = FileDocumentManager.getInstance().getFile(document);
+        assert virtualFile != null;
+        String fileName = virtualFile.getName();
+        String filePath = virtualFile.getPath();
+        int startLineOffset = document.getLineStartOffset(startLine);
+        int endLineOffset = document.getLineEndOffset(endLine);
+        final String selectedText = document.getText().substring(startLineOffset, endLineOffset);
+        String newText = FileUtils.addWriteSpace(text, selectedText);
+        VirtualFile rightFile = LocalFileSystem.getInstance().refreshAndFindFileByPath(filePath);
+        Path path = Paths.get(PathManager.getTempPath() + File.separator + DiffUtils.TEMP_DIR_NAME);
+        Files.createDirectories(path);
+        var tempFileName = FileUtils.getName(fileName,"") + "_" + DateTimeFormatter.ofPattern("yyyyMMddHHmmss").format(LocalDateTime.now()) + "." + virtualFile.getExtension();
+        String tempFilePath = PathManager.getTempPath() + File.separator + DiffUtils.TEMP_DIR_NAME + File.separator + tempFileName;
+        FileUtils.replaceContentInFile(filePath, tempFilePath, selectedText, newText, startLine, endLine);
+        File leftFileTemp = new File(tempFilePath);
+        VirtualFile leftFile = LocalFileSystem.getInstance().refreshAndFindFileByPath(leftFileTemp.getCanonicalPath());
+        assert leftFile != null;
+        DiffContent leftContent = DiffContentFactory.getInstance().create(project, leftFile);
+        assert rightFile != null;
+        DiffContent rightContent = DiffContentFactory.getInstance().create(project, rightFile);
+        String leftFileTitle = leftFile.getName() + DiffUtils.LEFT_TITLE;
+        String rightFileTitle = rightFile.getName() + DiffUtils.RIGHT_TITLE;
+        SimpleDiffRequest simpleDiffRequest = new SimpleDiffRequest(DiffUtils.DIFF_TITLE, leftContent, rightContent, leftFileTitle, rightFileTitle);
+        simpleDiffRequest.putUserData(DiffUserDataKeys.SCROLL_TO_LINE, Pair.create(Side.RIGHT, startLine));
+        simpleDiffRequest.putUserData(DiffUserDataKeys.PREFERRED_FOCUS_SIDE, Side.RIGHT);
+        simpleDiffRequest.putUserData(DiffUtils.DIFF_FILENAME, fileName);
+        simpleDiffRequest.putUserData(DiffUtils.DIFF_FILEPATH_LEFT, leftFile.getPath());
+        simpleDiffRequest.putUserData(DiffUtils.DIFF_FILEPATH_RIGHT, rightFile.getPath());
+        simpleDiffRequest.putUserData(DiffUtils.DIFF_FILE_UNIQUE_ID, DiffUtils.DIFF_ID);
+        DiffUtils.closeDiffViewIfAlreadyOpened(project);
+        diffManagerInstance.showDiff(project, simpleDiffRequest, DiffDialogHints.DEFAULT);
+    } catch (Exception e) {
+        throw new RuntimeException("Open diff failed");
```

Comment:
Log down the cause as well - `throw new RuntimeException("Open diff failed", e);`